Add support for aarch64 and some improvements

juanje opened the pull request.

juanje committed:
ostree.sh: remove extra verbosity

juanje committed:
ostree.sh: don't remove the VM and show the artifact path

juanje committed:
ostree.sh: support rhel8.4, c8s for amd64 and arm64

- Simplfy the supported distros and workarounds.
- Remove the HTTPBoot (not supported by aarch64).
- Make the URL arch indepemendent.

juanje committed:
Fix missing dirs at /var due systemd unit not enabled on c8s packages

rasibley commented on ostree.sh:
```diff
@@ -405,9 +375,10 @@ sudo virt-install  --name="${IMAGE_KEY}"\
-                   --pxe \
-                   --boot uefi,loader_ro=yes,loader_type=pflash,nvram_template=/usr/share/edk2/ovmf/OVMF_VARS.fd,loader_secure=no \
+                   --os-variant "${OS_VARIANT}" \
+		   --location "${BOOT_LOCATION}" \
```
one minor nitpick, looks like the indenting is off here

rasibley commented on ostree.sh:
```diff
@@ -46,26 +40,23 @@ sudo mkdir -p /etc/osbuild-composer/repositories
 
 # Set os-variant and boot location used by virt-install.
 case "${ID}-${VERSION_ID}" in
-    "rhel-8.3")
```
Why remove 8.3 case here, isn't ostree.sh supported on 8.3 ?

juanje committed:
Fix ssh private key too open

juanje committed:
Copy rhel-8.4 internal sources and add aarch64 repos

juanje committed:
Add some missing dependencies

juanje committed:
Save the c8s release files to restored later

To be able to re-run the script, the files need
to be restored at the end.

juanje committed:
Minor style fixes

juanje committed:
Show the OS version before to change the release file

juanje committed:
Remove the --noautoconsole for better local debugging

juanje committed:
Fix OSTREE_REF. It needs to be rhel/8/... It's hardcoded

juanje committed:
Clean previous repo cache and files

juanje committed:
Some improvements

- Remove sudo as the script is going to be exceuted with sudo.
- Remove tabs and fix indentations.
- Remove the upgrade and the ansible tests.
- Add some verbose to the copy files.
- Add check for the OS release via SSH after the installation is done.

juanje committed:
Without the sudo, we need to kill the process, not its parent

juanje committed:
Remove unused variable and some old comments
